graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/builtins/NumberFunctionBuiltins.java

```diff
@@ -134,6 +134,11 @@ protected boolean isFinite(@SuppressWarnings("unused") int arg) {
             return true;
         }
 
+        @Specialization
+        protected boolean isFinite(@SuppressWarnings("unused") long arg) {
+            return true;
+        }
+
```

Comment:
This is a correct specialization for `long` values, but you should also exclude `long` values from `isFinite(Object arg)` specialization (i.e. change it to `@Specialization(guards = {"!isNumber(arg)", "!isNumberLong(arg)"})`).

Note that a specialization is activated when a matching value is encountered. When additional values are encountered then the active specializations are tested first and new specializations are activated when none of the active specializations can be used only. So, the following test-case is broken in your case:
```
Number.isFinite('foo'): false
Number.isFinite(longValue): false
```
`isFinite(Object)` is activated on the first line and this specialization is taken on the second line as well because this specialization allows `Long` values.